### DISCUSSION TITLE
API endpoint per gli acquisti da registrare

### DISCUSSION THREAD

**Discussion by massi82** [fattureincloud/api#377] [Ideas]
Salve a tutti, 

non riesco a trovare l'endpoint in python che mi permetta di fare il retrieve di delle fatture da registrare sotto gli Acquisti. Per capirci, i documenti che visualizzo cliccando sulla voce nello screenshot allegato. Grazie in anticipo per l'aiuto.
![image](https://github.com/fattureincloud/api/assets/14828435/3cb5ef6c-40c7-4bef-a287-33269628d357)

**Comment by emajo** [comment]
Ciao @massi82, attualmente non disponiamo di un endpoint per recuperare le fatture da registrare.

Grazie,
Emanuele.

**Reply by massi82** [reply]
Ciao @emajo, grazie per la risposta... è previsto un rilascio di questo endpoint? o esistono workaround/suggerimenti per superare il problema?

Grazie mille in anticipo.

**Reply by BluePixel33** [reply]
Salve, ci sono novità sull'implementazione dell'endpoint per ricavare gli acquisti da registrare?

**Reply by Idealmedica** [reply]
Buongiorno, anche io sono interessato a sapere se c'è una data di rialscio di questa funzionalità, grazie

**Reply by valmoz** [reply]
Buongiorno,
vorremmo chiedervi gentilmente di andare un po' più nel dettaglio delle vostre esigenze per quanto riguarda questa funzionalità, in modo tale da capire se ci stiamo muovendo nella direzione giusta. Ci piacerebbe sapere qual è la necessità che state cercando di soddisfare, cosa vorreste implementare e cosa vi aspettereste dalle nostre API. 
Ovviamente, nel caso in cui non vi sentiste sicuri nello scrivere in questa discussione pubblica le vostre necessità, sentitevi liberi di contattare il nostro supporto clienti (allegando questa discussione per fornire un po' di contesto).

Grazie
Mauro

**Reply by Idealmedica** [reply]
Buongiorno, nel mio caso sarebbe utile avere tutti i riferimenti esistenti per le altre fatture di acquisto, ovvero dati fornitore, righe (e totali) contenuti in fattura, scadenza inserita da fornitore e metodo pagamento presente (con IBAN se presente).
Per i centri di costo e/o categorie nel mio caso non sarebbe un problema ma immagino per altri utenti potrebbe essere utile riuscire a modificarli tramite API

**Reply by valmoz** [reply]
@massi82 @BluePixel33 voi siete ancora interessati a questa funzionalità?

**Reply by andry80pa** [reply]
Io sono interessato sarebbe molto utile

**Reply by valmoz** [reply]
Buongiorno @andry80pa  potrebbe descriverci un po' più nel dettaglio la richiesta, rispondendo alla nostra [domanda del 27 maggio](https://github.com/fattureincloud/api/discussions/377#discussioncomment-9569776), in modo tale da capire un po' meglio le vostre esigenze.

**Reply by andry80pa** [reply]
Buongiorno,
si certo in questo momento chiamando l' API ListReceivedDocuments ( sto utilizzando l' SDK C# ) e settando come type "expense" ricevo i dati solo delle fatture registrate. Quello che sarebbe molto utile è invece ricevere anche le fatture ricevute non ancora registrate quindi quelle che nel backoffice si trovano sotto questa voce:

![screen](https://github.com/user-attachments/assets/cc4260c6-b12f-4260-89c0-8050c8cb5ab0)

che al momento è impossibile recuperare.
Se servono altre informaizioni ditemi pure.
Grazie

**Reply by andry80pa** [reply]
Buongiorno, scusate ci sono novità al riguardo? Anche solo sapere se effettivamente è in sviluppo o meno.
Grazie!

**Reply by andry80pa** [reply]
Buongiorno, scusate nuovamente, siccome in assenza di questa funzione devo muovermi diversamente con lo sviluppo desideravo solo sapere se è prevista o no, perchè se no procedo in maniera alternativa.
Grazie.

**Reply by valmoz** [reply]
Buongiorno,
non posso risponderle al momento in quando la sezione è attualmente sotto sviluppo attivo. Ci potrebbero essere novità nel prossimo futuro a riguardo, ma non ho visibilità di come cambierà questa sezione quindi non posso darle aggiornamenti in positivo o negativo.
Nel frattempo le consiglio di procedere in maniera alternativa.

Grazie
Mauro

**Reply by Idealmedica** [reply]
Buongiorno a tutti, per le fatture ricevute (a prescindere che siano registrate o meno) c'è un endpoint specifico oppure bisogna attendere lo sviluppo dell'endpoint delle fatture da registrare e poi andarle a registrare tramtie API?

**Reply by LuisOfCourse** [reply]
> Buongiorno a tutti, per le fatture ricevute (a prescindere che siano registrate o meno) c'è un endpoint specifico oppure bisogna attendere lo sviluppo dell'endpoint delle fatture da registrare e poi andarle a registrare tramtie API?

Per ottenere le fatture registrate e solo quelle puoi fare una chiamata come descritto qui:
[API Reference - Fatture in Cloud](https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/received_documents)
